features/collections.feature

```diff
@@ -116,7 +116,31 @@ Feature: Collections
     And the _site directory should exist
     And the "_site/puppies/fido.html" file should exist
 
-  Scenario: Hidden collection with document with future date
+  Scenario: Access rendered collection with future dated document via Liquid
+    Given I have a _puppies directory
+    And I have the following documents under the puppies collection:
+      | title  | date       | content             |
+      | Rover  | 2007-12-31 | content for Rover.  |
+      | Fido   | 2120-12-31 | content for Fido.   |
+    And I have a "_config.yml" file with content:
+    """
+    collections:
+      puppies:
+        output: true
+    """
+    And I have a "index.html" page that contains "Newest puppy: {% assign puppy = site.puppies.last %}{{ puppy.title }}"
+    When I run jekyll build
+    Then I should get a zero exit status
+    And the _site directory should exist
+    And I should see "Newest puppy: Fido" in "_site/index.html"
```

Comment:
Why do we return a future document here?

Reply:
I've documented the behavior in `_docs/collections.md`

This is based on https://github.com/jekyll/jekyll/pull/6506#issuecomment-340670610 and  https://github.com/jekyll/jekyll/pull/6506#issuecomment-340675231 by @pathawks.
You can iterate through *future docs* but won't be output / written to disk

Reply:
If this a breaking-change in behavior, we can ship this in `4.0`.. no biggie

Reply:
@pathawks use case with future events makes sense. 
Hugo's `date`, `publishsate` and `expirydate` is way more flexible.

Reply:
If Pat Hawks use case is possible on `master` then this PR is safe to ship in `3.x` otherwise its probably a `4.0`

Reply:
On `master`, future docs are rendered and can be accessed, but are not written to disk. :+1: